### DISCUSSION TITLE
Password for postgres to be changed

### DISCUSSION THREAD

**Discussion by rahulbeelur** [plausible/analytics#2928] [Self-Hosted Support]
Is there a way to initially provide with custom db, user and password for postgres during installation. I am using self-hosted docker installation.
If I try to give custom user and password, I get an error Password Authentication failed for user "postgres" even though I want to change my user which I provided with environment variable POSTGRES_USER. Why does db get created only with superuser postgres?

**Comment by ruslandoga** [comment]
👋 @rahulbeelur 

Are you using `docker-compose`? If so, can you please share your config for `plausible_db` and `plasusible` services?

**Reply by rahulbeelur** [reply]
@ruslandoga Hello,

Yes

  plausible_db:
        # supported versions are 12, 13, and 14
        image: postgres:14-alpine
        restart: always
        ports:
            - **** : ****
        volumes:
            - db-data:/var/lib/postgresql/data
        environment:
            - POSTGRES_DB=p$POSTGRES_DB
            - POSTGRES_USER=$POSTGRES_USER
            - POSTGRES_PASSWORD=$POSTGRES_PASSWORD
            
 plausible:
        image: plausible/analytics:v1.5
        restart: always
        command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
        depends_on:
            - plausible_db
            - plausible_events_db
            - mail
        ports:
            - ****  :  ****
        env_file:
            - plausible-conf.env

volumes:
    db-data:
        driver: local
    event-data:
        driver: local
        
        
This is the file which was present once cloned - git clone https://github.com/plausible/hosting

**Reply by ruslandoga** [reply]
Just to check, have you updated the `DATABASE_URL` in `plausible-conf.env`?

**Reply by ruslandoga** [reply]
If not set, it defaults to [`postgres://postgres:postgres@plausible_db:5432/plausible_db`](https://github.com/plausible/analytics/blob/16c4b9e3eb991399ec1c000d02364a75fed06a28/config/runtime.exs#L57), the format is

```
<scheme, doesn't matter>://<username>:<password>@<hostname>:<port>/<database>
```

[[docs]](https://plausible.io/docs/self-hosting-configuration#database)

**Reply by rahulbeelur** [reply]
No as the database is on the same machine. Will it not be an issue?

**Reply by ruslandoga** [reply]
If

```
$POSTGRES_DB
$POSTGRES_USER
$POSTGRES_PASSWORD
```

are not

```
plausible_db
postgres
postgres
```

and you haven't changed `DATABASE_URL` in `plausible-conf.env` I think it will be an issue :)

**Reply by rahulbeelur** [reply]
@ruslandoga 

example - 
server ip - 12.23.34.45

docker-compose.yaml - 

plausible_db:
# supported versions are 12, 13, and 14
image: postgres:14-alpine
restart: always
ports:
- 1234 : 5432
volumes:
- db-data:/var/lib/postgresql/data
environment:
- POSTGRES_DB=abc
- POSTGRES_USER=def
- POSTGRES_PASSWORD=xyz

plausible:
image: plausible/analytics:v1.5
restart: always
command: sh -c "sleep 10 && /entrypoint.sh db createdb && /entrypoint.sh db migrate && /entrypoint.sh run"
depends_on:
- plausible_db
- plausible_events_db
- mail
ports:
- **** : ****
env_file:
- plausible-conf.env

volumes:
db-data:
driver: local
event-data:
driver: local


DATABASE_URL=postgres://def:xyz@12.23.34.45:1234/abc

Is it postgres://user:password@server-ip:port-mapped-to/$POSTGRES_DB ?


I get the following error - 

Error - * (MatchError) no match of right hand side value: {:error, "connection not available and request was dropped from queue after 2000ms. This means requests are coming in and your connection pool cannot serve them fast enough. You can address this by:\n\n  1. Ensuring your database is available and that you can connect to it\n  2. Tracking down slow queries and making sure they are running fast enough\n  3. Increasing the pool_size (although this increases resource consumption)\n  4. Allowing requests to wait longer by increasing :queue_target and :queue_interval\n\nSee DBConnection.start_link/2 for more information\n"}

**Reply by rahulbeelur** [reply]
If provided with - 

postgres://def:xyz@localhost:1234/$POSTGRES_DB

I get the following error - 

Failed to connect: ** (DBConnection.ConnectionError) tcp connect (localhost:1234): connection refused - :econnrefused

**Reply by ruslandoga** [reply]
Please check that you have network access to `plausible_db` container from `plausible` container.

Have you tried

```
DATABASE_URL=postgres://def:xyz@localhost:5432/abc
```

or

```
DATABASE_URL=postgres://def:xyz@localhost:1234/abc
```

in `plausible-conf.env`?

**Reply by rahulbeelur** [reply]
Network access is fine as I tried to ping.


1. When I run

```
DATABASE_URL=postgres://def:xyz@localhost:1234/abc
```

I get the following error
- Postgrex.Protocol (#PID<0.173.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (localhost:1234): connection refused - :econnrefused

2. When I try this
```
DATABASE_URL=postgres://def:xyz@localhost:5432/abc
```

Postgrex.Protocol (#PID<0.173.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (localhost:5432): connection refused - :econnrefused

**Reply by rahulbeelur** [reply]
@ruslandoga 
the user which we provide should the role as superuser for it first already be created and then we use it? Normally it should create a super user with the user provided.

**Reply by ruslandoga** [reply]
I don't think roles matter much. The error indicates a network problem. Can you please try 

```
DATABASE_URL=postgres://def:xyz@plausible_db:5432/abc
```

and 

```
DATABASE_URL=postgres://def:xyz@plausible_db:1234/abc
```
